File: data/src/main/java/pl/mobly/data/AppInfoMapper.java
package pl.mobly.data;


import android.content.pm.ApplicationInfo;

import java.util.ArrayList;
import java.util.List;

import pl.mobly.domain.DomainApplicationInfo;

public class AppInfoMapper {

    static List<DomainApplicationInfo> mapToDomain(List<ApplicationInfo> info) {
        List<DomainApplicationInfo> domainApplicationInfos = new ArrayList<>();
        for (ApplicationInfo applicationInfo : info) {
            domainApplicationInfos.add(new DomainApplicationInfo(applicationInfo.name));
        }
        return domainApplicationInfos;
    }
}

File: data/src/main/java/pl/mobly/data/AppInfoMapper.kt
package pl.mobly.data


import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import pl.mobly.domain.DomainApplicationInfo
import java.util.*

object AppInfoMapper {

    internal fun mapToDomain(info: MutableList<ApplicationInfo>, packageManager: PackageManager): List<DomainApplicationInfo> {
        val domainApplicationInfos = ArrayList<DomainApplicationInfo>()
        for (appInfo in info) {
            domainApplicationInfos.add(DomainApplicationInfo(appInfo.loadLabel(packageManager).toString(), appInfo.packageName, appInfo.icon))
        }
        return domainApplicationInfos
    }
}
